### DISCUSSION TITLE
Why is my average loss curve continously increasing after 10,000 steps

### DISCUSSION THREAD

**Discussion by KakoozaJerry** [coqui-ai/TTS#838] [General Q&A] [answered]
Hello guys, I am trying to understand why model model seems to have it's average loss going totally up at about 10k steps despite initially going down for the strat. I have trained up to 102 k steps and here is my tensorboard : 

[https://tensorboard.dev/experiment/eJhaY5qDSemwCLRoaHVAFg/#scalars](https://tensorboard.dev/experiment/eJhaY5qDSemwCLRoaHVAFg/#scalars)

My dataset is using data worth 2 hours and 55 minutes of Audio

Am training using Tacotron2 with a batch size of 32 and  "gradual_training": [[0, 6, 32], [10000, 4, 32], [50000, 3, 32], [100000, 2, 32]] , I also turned off the trim_audio silences at the start. Set it to false

Please if anyone can help I would be grateful

**Comment by erogol** [comment]
The link above does not show any figures

**Reply by erogol** [reply]
Also if you are training the model from scratch 2 hours is not enough to get high-quality output. You should try fine-tuning one of the released models.

**Reply by KakoozaJerry** [reply]
> The link above does not show any figures

Oh sorry, I have edited it . Thank you

**Reply by KakoozaJerry** [reply]
> Also if you are training the model from scratch 2 hours is not enough to get high-quality output. You should try fine-tuning one of the released models.

Yes, I understand but then even with 2 hours, I thought I would atleast have a loss curve going down but then this one specifically goes down for a few steps then it exponentially goes up. I will also look into training on a pre existing model, just wondering if it will work since the language am building on has no model yet. Its Luganda

**Comment by erogol** [comment, marked as answer]
It is overfitting. You can try to increase the dropout rates or use a smaller model. 

If you finetune, you can try to finetune with a small learning rate with your dataset. 

how does the model output sound?

**Reply by KakoozaJerry** [reply]
Okay @erogol , thanks for the advise. So the output sounds like this  https://drive.google.com/drive/folders/1n8_38GH2SBZjBxNgD616LD4X-xYWsIiO?usp=sharing and sofar its the kind of robotic sound I think which is not desired and its not totally good yet. Also is there a way to give the voice more power / Volume for it to be heard well

**Reply by KakoozaJerry** [reply]
Maybe also , what are the good values of dropout rate you would suggest

**Reply by erogol** [reply]
That I don't know. You should probably start from something really high and lower until it works

**Reply by erogol** [reply]
maybe go like 0.8 -> 0.3

**Reply by KakoozaJerry** [reply]
Okay @erogol , and to set this dropout , do I use the -dropout_p flag

**Reply by erogol** [reply]
No you need to set it in the model config. But again 2h is not most probably enough for a perfect model. Just to see the expectation.

**Reply by KakoozaJerry** [reply]
Sure, am also trying to see if I can get a few more hours.  Maybe is there a way to give the voice more power / Volume for it to be heard well.  Does adjusting the power="1.5" to power="3" or "4" help?

**Reply by KakoozaJerry** [reply]
@erogol , I cannot seem to find the right parameter to use for dropout rate in Tacotron 2. I am only seeing it within GlowTTS

**Reply by erogol** [reply]
dropout rates for tacotron are not listed in the config. We need to expose them in the next version for such use cases. Until then you can set it in the source code before you train.

**Reply by KakoozaJerry** [reply]
Alright, @erogol , could you please direct me to which file in the source code and how i can do this. Will be so grateful, thanks

**Reply by erogol** [reply]
https://github.com/coqui-ai/TTS/blob/7f388f26e36c4af0606b9d99b2fb8736430fae4f/TTS/tts/layers/tacotron/tacotron2.py#L175

**Reply by KakoozaJerry** [reply]
Thanks alot @erogol , should the attention and decoder dropouts all have to be changed equally to have the same value or they can have different values?
